app/connectors/EventReportingConnector.scala

```diff
@@ -129,7 +132,7 @@ class EventReportingConnector @Inject()(
   }
 
   def submitReportEvent20A(pstr: String, ua: UserAnswers, version: String)
-                          (implicit headerCarrier: HeaderCarrier): Future[Unit] = {
+                          (implicit headerCarrier: HeaderCarrier): Future[Result] = {
```

Comment:
Change submitReportEvent20A return type to Future[HttpResponse] and handle via controller.

Reply:
After discussion with Pavel, these comments will be addressed in [PODS-8697](https://jira.tools.tax.service.gov.uk/browse/PODS-8697) as this is a substantial refactor which will require QA testing again.